clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java

```diff
@@ -665,7 +665,8 @@ private boolean processNextQueuedRemoteTask() {
             final RemoteClusterControllerTask task = remoteTasks.poll();
             log.finest(() -> String.format("Processing remote task of type '%s'", task.getClass().getName()));
             task.doRemoteFleetControllerTask(context);
-            if (!task.hasVersionAckDependency()) {
+            // We cannot introduce a version barrier for tasks when we're not the master (and therefore will not publish new versions).
+            if (!isMaster() || !task.hasVersionAckDependency()) {
```

Comment:
A better test than !isMaster() may be that the task has failed?

Reply:
Yes, good point. I'll do this in a follow-up PR after I create a PR for temporarily disabling the version ACK dependency for set-node-state tasks.

Need to do some more thinking about how the orchestrator will interact with the controller during leader elections. Not entirely comfortable with how threads will be blocked during the election grace period. The simplest solution might be to just bounce requests with "leader not known" failures if we're in the grace period. The client should then retry round-robin until it ends back up on the leader controller, eventually after its grace period has expired and it's a "proper" leader. Should also consider an explicit request timeout mechanism to avoid hanging threads for unbounded times if distributors in the cluster are failing to converge on a particular state.

Reply:
Bouncing w/retries should work already.

timeout sounds right imo, but there should also be timeout on server-side.

There was also some mentioning of filled up thread pools, which should also be looked into - or perhaps that works as expected?

Reply:
To clarify, my suggestions were all for changes in the behavior of the _cluster controller_, not the orchestrator. I.e. the request bounces and timeout handling would be done server-side. We already check pending tasks per controller tick, so straight forward to add an ACK deadline to them.

The thread pools, as far as I understand it, are filled up because they use a thread-per-request model. They explicitly block waiting for the completion of the `RemoteClusterControllerTask` instance they have scheduled. Prior to the fix in this PR, requests towards follower controllers would not have their tasks completed. In that respect, it's expected behavior from the pools.